### PR TITLE
Cache display name

### PR DESCRIPTION
**Pull request by CarlSchwan**
This should save some queries in the share backend when displaying the
owner and it's not important if the display name is 10 minutes outdated
as it is very rare that this gets changed.

image

### CODE REVIEW

File: apps/files_sharing/lib/Cache.php [hunk 1]
```diff
@@ -46,27 +47,22 @@
...
-
-	private $rootUnchanged = true;
-
-	private $ownerDisplayName;
-
-	private $numericId;
+	private ICacheEntry $sourceRootInfo;
+	private bool $rootUnchanged = true;
+	private ?string $ownerDisplayName = null;
+	private int $numericId;
```
Comment (CarlSchwan) [inline apps/files_sharing/lib/Cache.php:55]:
```suggestion
    /** @var int */
	private $numericId;
```

Let's revert this change

**Comment by juliushaertl** [comment]
Makes sense. It could actually be invalidated within the existing change detection that triggers the update of the oc_account entries through events, but since it is just used in the sharing related code, fine with me for now. Maybe we can add a todo for that.

https://github.com/nextcloud/server/blob/b90e657ac7b0760aecffeadf046908131053d6bf/lib/private/User/User.php#L557

**Comment by CarlSchwan** [comment]
@icewind1991 looks like we had the same idea https://github.com/nextcloud/server/pull/32050 :)

Moving this to another class makes it a bit more reusable and makes dependencies injections and event listening works.

**Comment by blizzz** [comment]
> @icewind1991 looks like we had the same idea #32050 :)
> 
> Moving this to another class makes it a bit more reusable and makes dependencies injections and event listening works.

Is this PR then obsolote, or complementary? Does need adjustements now that #32050 is merged?

**Comment by CarlSchwan** [comment]
> > @icewind1991 looks like we had the same idea #32050 :)
> > Moving this to another class makes it a bit more reusable and makes dependencies injections and event listening works.
> 
> Is this PR then obsolote, or complementary? Does need adjustements now that #32050 is merged?

It's complementary and I rebased it already

**Review by blizzz — APPROVED** [review]
LGTM

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
Backports?
